### ISSUE TITLE
docker ps hangs

### ISSUE THREAD

**Issue by justin8** [docker/docker#18013]
We recently upgraded a large number of servers from docker 1.4 to 1.8.3, and now several servers every week begin to hang when running `docker ps`. Requiring a reboot of the entire server to resolve; restarting the daemon alone does not resolve the issue.

`docker version: Docker version 1.8.3, build f4bf5c7`
```
docker info:
Docker version 1.8.3, build f4bf5c7
ubuntu@docker-gce-ae-230-prod:~$ docker info
Containers: 1663
Images: 1284
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 4610
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.2.0-c9
Operating System: Debian GNU/Linux 8 (jessie)
CPUs: 8
Total Memory: 51.11 GiB
Name: docker-gce-ae-230-prod
ID: V6UJ:OGDC:WFTX:C6JC:RV6T:7DQB:JEC5:GFXG:G7AS:AXQ5:6JXI:PNR5
Username: cloud9deploy
Registry: https://index.docker.io/v1/
```
`uname -a: Linux XXXXXXX 4.2.0-c9 #1 SMP Wed Sep 30 16:14:37 UTC 2015 x86_64 GNU/Linux`

Running on GCE.

**Comment by kunalkushwaha** [comment]
facing similar issue with v1.9.0 release with Consul. 
In my case, trying to setup swarm cluster with Consul as discovery backend.

Once, node is added, ``docker ps`` or  ``docker ls`` commands hangs and timeout takes almost 15~20 min.

**Comment by thaJeztah** [comment]
Has this been improved in docker 1.9.1? I know some changes were made in this area recently (e.g. https://github.com/docker/docker/issues/17720).

**Comment by yeasy** [comment]
With docker 1.11.2, still find this problem.

**Comment by thaJeztah** [comment]
@yeasy do you have some more information about your case; what were the steps after which it hung, how many containers/images, what platform/distro and graph driver are you using?

**Comment by yeasy** [comment]
@thaJeztah 
Yes.

dist: ubuntu 14.04.1 
kernel: 3.13.0-32-generic
storage: aufs
Docker: 1.11.2
containers: 10-20

scenarios: run docker exec for many times.

The happening is random, with low cpu/mem/disk usage.

Thanks!

**Comment by Nomon** [comment]
Happening for me also on 1.11.2, trying 1.12-rc2 next. Here are the system details where it happened:

```
ubuntu@ip-172-18-36-100:~$ uname -a
Linux ip-172-18-36-100 3.13.0-88-generic #135-Ubuntu SMP Wed Jun 8 21:10:42 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
ubuntu@ip-172-18-36-100:~$ sudo docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

```

full strace of docker ps
https://gist.github.com/Nomon/8d515046db5ed33eb08f304d78a6a194

**Comment by yeasy** [comment]
I found under `/run`, docker creates lots of fd, while not release after the docker exec.
Maybe this is the root cause.

**Comment by cpuguy83** [comment]
@yeasy There was a bug in 1.11.0 that did this, should not be a problem now.

**Comment by cpuguy83** [comment]
@Nomon Looks like you have a stacktrace of the docker client which is unfortunately not very helpful in this situation :(.
If you could provide the output (from the daemon side logs) after sending SIGUSR1 to the docker daemon this will give us what we need to help debug.

Thanks!

**Comment by Nomon** [comment]
@cpuguy83 I terminated that instance already, once it happen again I will do it. Thanks.

**Comment by thaJeztah** [comment]
closing because there's no activity, but happy to reopen if you're still having this issue, and have more information

**Comment by andyxning** [comment]
In Docker 1.12.0, we also encounter this `docker ps` hang. OS Debian 8 with Kernel 3.16.7-ckt20-1+deb8u4

**Comment by thaJeztah** [comment]
@andyxning do you have the same issue on docker 1.12.3? Various bug fixes went into 1.12.1, 1.12.2 and  1.12.3. If you still run into this issue in the current version (1.12.3 at time of writing), please open a new issue and provide more information. Commenting on closed issues may result in your comment being overlooked.

**Comment by andyxning** [comment]
​Ok, i will try 1.12.3 later. However, did we find what is the actual
reason for docker behave like this.​

=======
Ning Xie


2016-11-21 20:45 GMT+08:00 Sebastiaan van Stijn <notifications@github.com>:
